AppCenterCrashes/AppCenterCrashes/MSCrashes.mm

```diff
@@ -277,6 +279,22 @@ - (instancetype)init {
      * Setting this flag will let application crash on uncaught exceptions.
      */
     [[NSUserDefaults standardUserDefaults] registerDefaults:@{ @"NSApplicationCrashOnExceptions" : @YES }];
+
+    /*
+     * Exceptions on the main thread of a Cocoa application do not typically rise to the level of
+     * the uncaught exception handler because the global application object catches all such exceptions.
+     * See: https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/Exceptions/Concepts/UncaughtExceptions.html
+     */
+    if (![MSAppCenter isDebuggerAttached]) {
+      SEL selector = @selector(reportException:);
+      Method method = class_getInstanceMethod([NSApplication class], selector);
+      IMP implementation = class_getMethodImplementation([MSCrashes class], selector);
```

Comment:
What do we do with the original implementation? We should also execute the original implementation, it feels like people are expecting this method [to write the exception to the console](https://developer.apple.com/documentation/appkit/nsapplication/1428396-reportexception?language=objc).

Reply:
Good point!

Reply:
Calling the original one cause the crash with SIGILL (due to `NSApplicationCrashOnExceptions` setting I guess), and calling PLCR handler also aborts the execution, so calling the original one is problematic. I don't want to change this setting to break anything

Reply:
Thx for the clarification Ivan!

One suggestion: Can we rename the entry to the `info.plist` something that requires less technical knowledge. I suggest `kMSIsAppCenterCrashForwardingMacOsExceptionsAsCrashesKey`.
I'm not blocking on this one as it's the naming of a variable.

Reply:
If I understood correctly if we call the original method before PLCR handler it will crash depending on the setting but if we call it afterwards then it won't do anything since PLCR will stop the execution. OK we are stuck that means we'll have to document the limitation of this approach. The crash will still be logged in the console by the SDK correct log level activated anyways so it's not really an issue.

Reply:
Basically if the customer shouldn't call `reportException` themselves if this is used.

Reply:
@clpolet It isn't changed by this PR. Calling `reportException` crashes the application because the SDK sets that flag. I've already suggested get rid of this behavior and report handled exceptions instead.

Reply:
@ElektrojungeMS This name makes confusion because even if it disabled the SDK changes behavior to crash the application on exception (and it stored for next runs).